src/main/java/com/actiontech/dble/config/helper/GetAndSyncDbInstanceKeyVariables.java

```diff
@@ -86,89 +88,85 @@ public KeyVariables call() {
     private class GetDbInstanceKeyVariablesListener implements SQLQueryResultListener<SQLQueryResult<Map<String, String>>> {
         @Override
         public void onResult(SQLQueryResult<Map<String, String>> result) {
-            if (result.isSuccess()) {
-                ds.setTestConnSuccess(true);
-                keyVariables = new KeyVariables();
-                keyVariables.setLowerCase(!result.getResult().get(COLUMN_LOWER_CASE).equals("0"));
-
-                keyVariables.setAutocommit(result.getResult().get(COLUMN_AUTOCOMMIT).equals("1"));
-                keyVariables.setTargetAutocommit(SystemConfig.getInstance().getAutocommit() == 1);
-
-                String isolation = result.getResult().get(columnIsolation);
-                switch (isolation) {
-                    case "READ-COMMITTED":
-                        keyVariables.setIsolation(Isolations.READ_COMMITTED);
-                        break;
-                    case "READ-UNCOMMITTED":
-                        keyVariables.setIsolation(Isolations.READ_UNCOMMITTED);
-                        break;
-                    case "REPEATABLE-READ":
-                        keyVariables.setIsolation(Isolations.REPEATABLE_READ);
-                        break;
-                    case "SERIALIZABLE":
-                        keyVariables.setIsolation(Isolations.SERIALIZABLE);
-                        break;
-                    default:
-                        break;
+            boolean isSuccess = result.isSuccess();
```

Comment:
reduce the lock range and using the temp variable will  more easily than this change

Reply:
good idea